motionplan/tpSpaceRRT.go

```diff
@@ -0,0 +1,439 @@
+//go:build !windows
+
+package motionplan
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"math/rand"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/motionplan/tpspace"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+var (
+	defaultGoalCheck = 5  // Check if the goal is reachable every this many iterations
+	defaultAutoBB    = 1. // Automatic bounding box on driveable area as a multiple of start-goal distance
+
+	// whether to add intermediate waypoints.
+	defaultAddInt = false
+	// Add a subnode every this many mm along a valid trajectory. Large values run faster, small gives better paths
+	// Meaningless if the above is false.
+	defaultAddNodeEvery = 50.
+
+	// When getting the closest node to a pose, only look for nodes at least this far away.
+	defaultDuplicateNodeBuffer = 50.
+
+	// Don't add new RRT tree nodes if there is an existing node within this distance.
+	defaultIdenticalNodeDistance = 5.
+)
+
+// newTPSpaceMotionPlanner creates a newTPSpaceMotionPlanner object with a user specified random seed.
+func newTPSpaceMotionPlanner(
+	frame referenceframe.Frame,
+	seed *rand.Rand,
+	logger golog.Logger,
+	opt *plannerOptions,
+) (motionPlanner, error) {
+	if opt == nil {
```

Comment:
`plannerOptions` are specific to the motion being done and are carefully constructed by the planManager. Using some default set is worse than erroring, because we should never be in a situation where we are without one.